src/wp-admin/includes/class-plugin-upgrader.php

```diff
@@ -482,6 +482,8 @@ public function check_package( $source ) {
 			foreach ( $files as $file ) {
 				$info = get_plugin_data( $file, false, false );
 				if ( ! empty( $info['Name'] ) ) {
+					$info['File']          = $file;
+					$info['Slug']          = rtrim( basename( $file ), '.php' );
```

Comment:
I'm storing both of these despite only using the slug for now, as I'm not 100% sure if `WP_Upgrader::install_package()` needs the slug, or the filepath relative to `WP_PLUGIN_DIR`. Once this is known, I'll remove the unneeded one.